src/py/flwr/superexec/exec_servicer_test.py

```diff
@@ -104,3 +107,24 @@ def test_list_run_id(self) -> None:
         # Assert
         self.assertLess(abs(retrieved_timestamp - now().timestamp()), 1e-3)
         self.assertEqual(set(response.run_dict.keys()), {run_id})
+
+    def test_stop_run(self) -> None:
+        """Test StopRun method of ExecServicer."""
+        # Prepare
+        run_id = self.state.create_run(
+            "mock fabid", "mock fabver", "fake hash", {}, ConfigsRecord()
+        )
+        self.servicer.executor = MagicMock()
+        expected_run_status = RunStatus(Status.FINISHED, SubStatus.STOPPED, "")
+        self.servicer.executor.stop_run = lambda input_run_id: (
+            input_run_id == run_id
+        ) & self.state.update_run_status(input_run_id, new_status=expected_run_status)
+
+        # Execute
+        response = self.servicer.StopRun(StopRunRequest(run_id=run_id), Mock())
+        run_state = self.state.get_run(run_id)
+
+        # Assert
+        self.assertTrue(response.success)
+        if run_state is not None:
```

Comment:
should we assert if `run_state` is None? and leave the check below outside an if statement? `run_state` should never be None in this test

Reply:
Yes, you're right that `run_state` should never be None. I'll add a check for this. This `if`-statement is resolve the `mypy` "Item "None" of "Run | None" has no attribute "status"" error.